callbacks/callmethod.go

```diff
@@ -17,6 +17,9 @@ func callMethod(db *gorm.DB, fc func(value interface{}, tx *gorm.DB) bool) {
 				db.Statement.CurDestIndex++
 			}
 		case reflect.Struct:
+			if !db.Statement.ReflectValue.CanAddr() {
+				db.Statement.ReflectValue = reflect.New(db.Statement.ReflectValue.Type()).Elem()
```

Comment:
Modifying ReflectValue may produce unexpected results, there is no need to modify it here.
Actually I don't think it's necessary to implement it for not addressable model.

Reply:
just like #5990?
so we close this and merge 5990

Reply:
#5990 still has some issues, not re-reviewed since it hasn't been updated recently.

Reply:
Updated